Mage/src/main/java/mage/abilities/common/DealsDamageToOpponentTriggeredAbility.java

```diff
@@ -32,6 +32,7 @@ public DealsDamageToOpponentTriggeredAbility(Effect effect, boolean optional, bo
         super(Zone.BATTLEFIELD, effect, optional);
         this.onlyCombat = onlyCombat;
         this.setTargetPointer = setTargetPointer;
+        setTriggerPhrase("Whenever {this} deals " + (onlyCombat ? "combat" : "") + " damage to an opponent, ");
```

Comment:
This generates an extra space when `onlyCombat` is false. The space that's before "damage" needs to be moved to after "combat", matching other similar trigger phrases.